lib/fastlane/actions/resize_app_icon.rb

```diff
@@ -0,0 +1,120 @@
+module Fastlane
+  module Actions
+    # ResizeAction is based on work created and Copyright (c) 2014 Erik Sundin
+    # Original source may be found under the MIT license at https://github.com/eriksundin/appicon
+    class ResizeAppIconAction < Action
+      def self.run(params)
+        # fastlane will take care of reading in the parameter and fetching the environment variable:
+        Helper.log.info "Path to source icon: #{params[:iconPath]}"
+        Helper.log.info "Path to Images.xcassets: #{params[:assetsPath]}"
+
+        if params[:iconPath] and params[:assetsPath] and !params[:iconPath].empty? and !params[:assetsPath].empty?
+          validate_image_magick!
+
+          determine_icon! unless @icon = params[:iconPath]
+          determine_asset_catalog! unless @asset_catalog = params[:assetsPath]
```

Comment:
hey @calebhicks, im working on this PR a bit, and am curious about these two values (or method calls?) `determine_icon!` and `determine_asset_catalog!` as i am not seeing method definitions for these. 

do you mind just letting me know a little about what these lines do?